src/plugins/ui_stateful/clipboard.ts

```diff
@@ -109,6 +110,17 @@ export class ClipboardPlugin extends UIPlugin {
           this.status = "invisible";
         }
         break;
+      case "REPLICATE_CELLS_ABOVE":
+      case "REPLICATE_CELLS_ON_LEFT":
+        const zone = this.getters.getSelectedZone();
+        const copyTarget = ("specificTarget" in cmd && cmd.specificTarget) || [zone];
+        this.state = this.getClipboardState(copyTarget, "COPY");
```

Comment:
Okay so at present, I've pushed your suggestions which were on common grounds with the approach suggested 
by `rrahir`.
But I figure there's a contradiction or dilemma regarding introduction of specific commands as well as for using clipboard state directly like `laa`.

So I'm currently awaiting a final conclusion on desired approach from here on. Let me know which one of your approaches would be preferable and I'd then write my fix accordingly.
@rrahir @hokolomopo

Reply:
I think your current implementation is fine. The component stays dumb and do the minimum amount of things. And the plugin re-uses the `ClipboardCellState` object so the code is really simple. My only question would be wther the command should be handeld by the Clipboard or another plugin.

But let's ask another opinion @LucasLefevre

Reply:
First things first, I am not attached to any strategy right now, I think there are theoretical pros and cons everywhere.there are some situations where the components end up wise/not-that-dumb  as they are on this lien where they want to dispatch to a plugin but there is next to 0 value added to move the logic to a plugin, except "don't put it in the component". 
I personally have some issues with the aforementioned strategy that we'll overlook right now but I feel like the commands that we design seem to me soooooo specific that we could argue that they are "action-like" (since they mimic  a user will, clearly translate a UI will).
I have come upon the same conundrum recently when wanting to implement the equivalent of 'Ctrl+Enter'  shortcut while editing a cell (it also uses the clipboard state), so facing the same issue and I am really interested by the direction we want to take :-)

Reply:
Taking a step back - we currently need a "parent" command to unify all core changes that would be dispatched by the clipboard state so the command does make sense ...

Reply:
@rrahir Okay, so I moved the logic to plugin as suggested by @hokolomopo, but as far as it is concerned, I'll move it back to the component.

Also, I figure that the "parent" command would be dispatched by clipboard state. But I'm not sure how exactly it will help unifying things, since we already handle all the core changes in clipboard plugin... Can you please elaborate a bit about the way you want me to implement this?

Reply:
The crucial point about the "parent" command is that we want our action (in this case the cell replication) needs to be executed in a single history step (i.e., if I undo it, every change made by the action should be reverted). There is a rule of thumb for this.  each time you call `dispatch` from outside the model (e;g. `env.model.dispatch`) it will make one history step.
On the contrary, every dispatch made from inside a plugin will be part of the very same history step,

This means that from an action, we should dispatch _once and only once_!

Now looking at the solution we have here, we want to use a `ClipboardCellsState` but this object might (and most likely will) dispatch several times. So we need this object to be used inside a plugin to have all its changes in the same history step.

This means that we need a generic command to call from outside the model, (REPLICATE_xxx) and this command will be handled inside a plugin to use the clipboard state.

tl;dr your implementation is right ATM

Reply:
Oh okay, I thought you probably wanted to me implement something else apart from my current implementation. 
But, thanks for clarifying and technically explaining the dispatch and history relation. I'll keep this in mind from now on :+1: